foursight_core/run_result.py

```diff
@@ -370,6 +370,11 @@ def __init__(self, connections, name, init_uuid=None):
         # self.action_name is the function name of the action to link to check
         self.action = ''
         self.allow_action = False  # by default do not allow the action to be run
+        # Added prevent_action to allow the allow_action flag to be True, so action *may* *always*
```

Comment:
Yes but this situation is different/new - where a check runs/succeeds but we do NOT want to automatically run the associated action AND/BUT we ALSO want to ALLOW the user to MANUALLY run the associated action. I don't think there is an existing mechanism in place for this.

Reply:
So is it the case that `allow_action` is really `action_manually_requested` and that `prevent_action` is really `inhibit_automatic_action` such that the action would run if `action_manually_requested or (check_succeeded and not inhibit_automatic_action)` or something like that? Sometimes I think good naming helps make these things make more sense. I presume that if this logic is not correct, it's because I haven't named things in the way that best represents the distinctions you're trying to make.